src/pybind/mgr/dashboard/HACKING.rst

```diff
@@ -3,6 +3,142 @@ Ceph Dashboard Developer Documentation
+chapter gives an introduction on how to set up such a system for development,
+without the need to set up a full blown production environment. All in this
+chapter introduced options are based on a so called `vstart` environment. 
+
+.. note::
+
+  Every `vstart` environment needs Ceph `to be compiled`_ from its Github
+  repository, though Docker environments simplify that step by providing a
+  shell script that contains those instructions.
+
+.. _to be compiled: https://docs.ceph.com/docs/master/install/build-ceph/
+
+vstart
+~~~~~~
+
+"vstart" is actually a shell script in the `src/` directory of the Ceph
+repository (``src/vstart.sh``). It is used to start a single node Ceph
+cluster on the machine where it is executed. Several required and some
+optional Ceph internal services are started automatically when it is used to
+start a Ceph cluster. `vstart` is the basis for the three most commonly used
+development environments in Ceph Dashboard.
+
+Host vs Docker
+~~~~~~~~~~~~~~
+
+This document introduces you to three different development environments, all
+based on vstart. Those are:
+
+- vstart running on your host system
+
+- vstart running in a Docker environment
+
+  * ceph-dev-docker_
+  * ceph-dev_
+
+.. note::
+
+  Independently of which of these environments you will choose, you need to
+  compile Ceph in that environment. If you compiled Ceph on your host system,
+  you would have to recompile it on Docker to be able to switch to a Docker
+  based solution. The same is true vice versa. If you previously used a
+  Docker development environment and compiled Ceph there and you now want to
+  switch to your host system, you will also need to recompile Ceph (or
+  compile Ceph using another separate repository).
+
+Development environment on your host system
+...........................................
+
+- No need to learn or have experience with Docker, jump in right away
+- Limited amount of scripts to support automation (like Ceph compilation)
+- No pre-configured easy-to-start services (Prometheus, Grafana, etc)
+- Limited amount of host operating systems supported, depending on which
+  Ceph version is supposed to be used
+
+Development environment using Docker
+....................................
+
+- Some overhead in learning Docker if you are not used to it yet.
+
+- Both Docker projects provide you with scripts that help you getting started
+  and automate recurring tasks.
+
+- Both Docker environments come with partly pre-configured external services
+  which can be used to attach to or complement Ceph Dashboard features, like
+
+  - Prometheus
+  - Grafana
+  - Node-Exporter
+  - Shibboleth
+  - HAProxy
+
+Many of the Ceph Dashboard developers are SUSE employees. We therefore recommend
+you to use `ceph-dev-docker`, which is based on SUSE Tumbleweed. 
+
+vstart on your host system
+~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The vstart script is usually called from your `build/` directory like so:
```

Comment:
Please note that `HACKING.rst` uses ReStructuredText, not MarkDown:
```suggestion
The vstart script is usually called from your ``build/`` directory like so:
```

Reply:
I did create a HTML file which I've also checked multiple times, but here seem to be differences in the rendering between what I've used locally and how Github behaves.